### DISCUSSION TITLE
Exporting visualization as images

### DISCUSSION THREAD

**Discussion by kornel-at-swoop** [jtablesaw/tablesaw#1200] [Q&A]
Hello,

I need to export chart images in well known image formats such as PNG or SVG. Basically, I need to load data from CSVs from Java, visualize it, and then export it as chart images. 

The reason for this is that I need to show charts in a platform that does not support showing charts but does support showing images. I would like to call out to a java library to produce the visualizations and import the output as images into my platform.

Can tablesaw do that? If not, is there any java library that can produce nice visualizations of data and export them as images?

Any help would be greatly appreciated.

Thx
Kornel

**Comment by lwhite1** [comment]
Tablesaw uses JavaScript to create the images so it can’t do what you want
without user interaction, unless you can setup a JavaScript engine as part
of your pipeline.

There are some Java FX libraries that produce nice images, but you would
need to search them out.  I don’t recall their names
